root/static/scripts/edit/URLCleanup.js

```diff
@@ -4826,6 +4826,35 @@ const CLEANUPS: CleanupEntries = {
       return url;
     },
   },
+  'tsutaya': {
+    match: [new RegExp('^(https?://)?shop\\.tsutaya\\.co\\.jp/', 'i')],
+    restrict: [LINK_TYPES.mailorder],
+    clean: function (url) {
+      url = url.replace(/^(?:https?:\/\/)?shop\.tsutaya\.co\.jp\/cd\/product\/(\d+)(?:\/.*)?$/, 'https://shop.tsutaya.co.jp/cd/product/$1/');
```

Comment:
It removes subpaths `\/.*` only whereas everything `.*$` that comes after the numeric ID `\d+` can be ignored (while still writing with a final slash `/`):

```suggestion
      url = url.replace(/^(?:https?:\/\/)?shop\.tsutaya\.co\.jp\/cd\/product\/(\d+).*$/, 'https://shop.tsutaya.co.jp/cd/product/$1/');
```

Details: The numeric ID `\d+` is fully matched because the quantifier `+` is _greedy_ unless specified _lazy_ in JavaScript; Actually all quantifiers are; See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Quantifiers for details.